### PR TITLE
add mpi and openmp monitoring and control examples

### LINKED ISSUE #96: Hello World Example for MPI programs
Add an example on how to use variorum with an MPI program for power monitoring and capping across different architectures.

### LINKED ISSUE #207: Process safe MPI monitoring and capping examples

### PR DESCRIPTION
**Pull request by slabasan**
- rank 0 is responsible for monitoring and control
- add mpi package to github actions
- add simple mpi and openmp integration C examples

Fixes #96

### CODE REVIEW

**Comment by tpatki** [comment]
@slabasan 
As discussed, we will update this to a version that can demonstrate a multi-node setup where power can be printed/capped on each node, along with a high-level script to explain invocation with MPI/srun/lrun etc.

**Comment by slabasan** [comment]
@tpatki This is ready for review.

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -102,6 +120,8 @@ jobs:
             BUILD_TESTS: ON
             CMAKE_BUILD_TYPE: Release
             ENABLE_FORTRAN: OFF
+            ENABLE_MPI: OFF
+            ENABLE_OPENMP: OFF
```
Comment (tpatki) [inline .github/workflows/github-actions.yml:124]: Should these (MPI/OPENMP) be enabled for the Release version as well? Thoughts?

**Comment by tpatki** [comment]
@slabasan The PR looks good and is mostly good to go. I want to test it once at my end though.

I have a few questions regarding where the examples should be enabled by default (I think they are not enabled in the Release version)? Seems like the local host-configs enable MPI only, so wanted to discuss with you. 

 Also, I think we need to update the documentation to include a dependency on VeloC (rankstr) and GitHub link ( https://github.com/ECP-VeloC/rankstr.git). We should also add documentation on the MPICH dependency (and whether rankstr is supported by other MPI implementations).

**Review by tpatki — CHANGES_REQUESTED** [review]

**Comment by slabasan** [comment]
Thanks for the feedback @tpatki! The default build for Variorum is RelWithDebugInfo, and the MPI and OpenMP examples should be enabled for this build.

File: .github/workflows/github-actions.yml (hunk 1 above)
Comment (slabasan) [reply .github/workflows/github-actions.yml:124]: This is one of the test configs for CI, we do test Variorum's default build (RelWithDebugInfo) with MPI and OpenMPI on, as well as other variants of this too.

**Comment by tpatki** [comment]
LGTM! I think we can go ahead and merge. @slabasan

**Pull request merged**